src/components/split-button/split-button.pw.tsx

```diff
@@ -591,13 +591,13 @@ test.describe("Functional tests in a custom component", () => {
   }) => {
     await mount(<TwoButtonsWithWrapper />);
 
-    await splitToggleButton(page).nth(0).click();
+    await splitToggleButton(page).nth(0).click({ delay: 1000 });
```

Comment:
**suggestion**: what do you think about waiting until the button list or first button explicitly appears or disappears, and interacting with each button using specific locators, rather than relying on the keyboard API?

[Reducing reliance on hardcoded timeouts](https://www.checklyhq.com/learn/playwright/waits-and-timeouts/) could enhance test robustness and minimise the risk of flakiness caused by random latencies. By waiting for well-defined conditions, the test behaviour could become more predictable, and we might also see reduced runtime in cases where elements update faster than expected.

```ts
await splitToggleButton(page).nth(0).click();

const buttonList = page.getByRole("list");
await buttonList.waitFor(); // waits for button list to be visible

const firstButton = buttonList.getByRole("button").first();
const secondButton = buttonList.getByRole("button").nth(1);
const thirdButton = buttonList.getByRole("button").last();

await expect(firstButton).toBeFocused();

await firstButton.press("Tab"); // Focuses specific element and then presses key

await expect(secondButton).toBeFocused(); 

await secondButton.press("Tab");

await expect(thirdButton).toBeFocused();

await thirdButton.press("Tab");

await expect(buttonList).toBeHidden(); // auto-retries until hidden
await expect(mainButton(page).nth(1)).toBeFocused();
```